lua/damagelogs/server/rdm_manager.lua

```diff
@@ -679,11 +679,45 @@ hook_Add("PlayerInitialSpawn", "PlayerInitialSpawn_RDM_Manager", function(ply)
     end)
 end)
 
+local function CheckRevive(ply)
+    if CR_VERSION and ply:IsRespawning() then
+        return true
+    elseif TTT2 and ply:IsReviving() then
+        return true
+    end
+    return false
+end
+
 hook_Add("PlayerDeath", "RDM_Manager", function(ply)
+
+    if GetRoundState() == ROUND_ACTIVE and (TTT2 or CR_VERSION) then
+        --Instant Respawn Cases
+        timer.Simple(1, function()
+            if ply:Alive() then return end
+        end)
+        -- Delayed Respawn Check
+        if CheckRevive(ply) then
+            net.Start("DL_Delayed")
+            net.Send(ply)
+
+            timer.Create("RDM_RespawnDelay_" .. ply:SteamID64(), 5, 1, function()
```

Comment:
Would 5 seconds be long enough for the respawn mechanics?

Reply:
Unfortunately role respawns vary and are often controlled by cvars so a flat 5 second timer may not work.
I'm not completely clear on what this is doing so I can't offer advice, really.

Reply:
From what I'm commonly seeing, there are 3 kinds of role respawns,
Instant: Has an effect on death and usually doesn't state the respawning state during that small instance  (Swapper)
Delayed: Respawns on a timer which usually doesn't last more than 5 +1 seconds (Zombie)
Conditional: Only respawns when a condition is fulfilled, which can range from happening instantly or never (Parasite)

The checks cover the first 2, while conditional has only 5-second grace before having to respond to reports.